vsutil/clips.py

```diff
@@ -22,20 +22,34 @@ def depth(clip: vs.VideoNode,
           range_in: Optional[Union[int, types.Range]] = None,
           dither_type: Optional[Union[types.Dither, str]] = None,
           ) -> vs.VideoNode:
-    """
-    A bit depth converter only using core.resize and Format.replace.
-    By default, outputs FLOAT sample type for 32 bit and INTEGER for anything else.
-
-    :param bitdepth:    Desired bits_per_sample of output clip.
-    :param sample_type: Desired sample_type of output clip. Allows overriding default FLOAT/INTEGER behavior. Accepts
-                        vapoursynth.SampleType enums INTEGER and FLOAT or their values, [0, 1].
-    :param range:       Output pixel range (defaults to input clip's range). See `Range`.
-    :param range_in:    Input pixel range (defaults to input clip's range). See `Range`.
-    :param dither_type: Dithering algorithm. Allows overriding default dithering behavior. See `Dither`.
-        Defaults to Floyd-Steinberg error diffusion when downsampling, converting between ranges, or upsampling full
-        range input. Defaults to 'none', or round to nearest, otherwise. See `_should_dither()` for more information.
-
-    :return: Converted clip with desired bit depth and sample type. ColorFamily will be same as input.
+    """A bit depth converter only using ``vapoursynth.core.resize()`` and ``vapoursynth.Format.replace()``.
+    By default, outputs ``vapoursynth.FLOAT`` sample type for 32-bit and ``vapoursynth.INTEGER`` for anything else.
+
+    >>> src_8 = vs.core.std.BlankClip(format=vs.YUV420P8)
+    >>> src_10 = depth(src_8, 10)
+    >>> src_10.format.name
+    'YUV420P10'
+
+    >>> src2_10 = vs.core.std.BlankClip(format=vs.RGB30)
+    >>> src2_8 = depth(src2_10, 8, dither_type=Dither.RANDOM)  # override default dither behavior
+    >>> src2_8.format.name
+    'RGB24'
+
+    :param clip:         Input clip.
+    :param bitdepth:     Desired `bits_per_sample` of output clip.
```

Comment:
Are the variable names supposed to be in single backticks? You do that in a lot of places (`bits_per_sample` here, `sample_type` and `clip` later, `frame` further down, but it doesn’t feel consistent to me.

Reply:
Makes it a bit clearer on the page when I'm refering to "a clip" vs. the input parameter _clip_, as single backticks just make them italics in rst. Subjective since there's no real style guide here, but I'm pretty sure I was consistent with how I did it. I always avoided referencing the parameters in the first line, or the one-sentence docstring summary, but would use them anywhere else. Here, `bits_per_sample` actually refers to the `vapoursynth.Format` parameter, not something from the function, so this is a bit different, but felt this didn't belong in in-line code either.

Reply:
I don’t have strong feelings about this, but I don’t think we’ll be able to keep it consistent. My intuition would have been to put all variable/parameter names in `code`, but it’s probably just a waste of time to think too hard about when to italicize something in docstrings. We’ll keep it like this.